### PR TITLE
Fix highlighting in database debug toolbar

### PR DESCRIPTION
**Pull request by vlakoff**
Highlighting wasn't working for multiple-word terms, as they were searched in the SQL string with nbsp entities instead of regular spaces.

This code could be optimized, though it's not crucial as it's a debug function.

### CODE REVIEW

**Comment by vlakoff** [comment]
Added a commit implementing replacement using regexp with word boundaries, to avoid overlaps / false positives with strings such as "OR", "IN"…

**Comment by sfadschm** [comment]
Can you add before/after schreenshots?

File: system/Database/Query.php [hunk 1]
```diff
@@ -406,10 +404,15 @@ public function debugToolbarDisplay(): string
...
-        }
+        $search = '/\b(?:' . implode('|', $highlightWords) . ')\b/';
+
+        $sql = preg_replace_callback($search, static function ($matches) {
+            return '<strong>' . str_replace(' ', '&nbsp;', $matches[0]) . '</strong>';
+        }, $sql);
+
+        $sql = str_replace(['(', ')'], ['<strong>(</strong>', '<strong>)</strong>'], $sql);
 
+        /** @noRector \Rector\CodeQuality\Rector\Return_\SimplifyUselessVariableRector */
```
Comment (paulbalandan) [inline system/Database/Query.php:415]: I would agree with Rector here to simplify the return. Besides, our coding standard also includes this rule to simplify the return when possible (although it is not taking effect here).

**Comment by vlakoff** [comment]
I confirmed the fix, but if ever you want screenshots of this toolbar content, you may have a look at #5136.

**Comment by sfadschm** [comment]
> I confirmed the fix, but if ever you want screenshots of this toolbar content, you may have a look at #5136.

There is unfortunately none of the key words changed here visible in the other PR. Would be great to see just one example what has been fixed in this PR, too.

**Review by paulbalandan — CHANGES_REQUESTED** [review]
Since this PR addresses an error in highlighting multiple-word identifiers, please add before/after screenshots, as @sfadschm suggested, that show there are indeed problems with the current and how this PR fixes them.

**Comment by vlakoff** [comment]
Before:
5129-before

After:
5129-after

**Comment by vlakoff** [comment]
Maybe we should add the ASC and DESC keywords?

**Comment by vlakoff** [comment]
Also, I'd suggest either highlighting commas in addition to parentheses, or not highlighting parentheses. Your pick?

**Comment by sfadschm** [comment]
I think adding the keywords would be good, but would rather remove the parentheses. But lets wait for other calls.

**Comment by kenjis** [comment]
I would like removing parentheses, adding ASC and DESC.

**Comment by paulbalandan** [comment]
Let's highlight only keywords, leaving out parentheses and other punctuation marks.

**Comment by vlakoff** [comment]
* Removed the highlighting of parentheses
* Added ASC and DESC keywords
* Reduced the number of commits by removing the commits related to the Rector warning

**Review by paulbalandan — APPROVED** [review]

**Comment by paulbalandan** [comment]
Thank you, @vlakoff

**Pull request merged**